repository/src/lib.rs

```diff
@@ -52,26 +52,56 @@ impl<T: RawRepository> DistributedRepository<T> {
         Ok(Self { raw })
     }
 
-    /// Initializes the genesis repository from the genesis working tree.
+    /// Initializes the genesis repository from the genesis working tree,
+    /// leaving a genesis header.
     pub async fn genesis(&mut self) -> Result<(), Error> {
         unimplemented!()
     }
     /// Returns the block header from the `finalized` branch.
     pub async fn get_last_finalized_block_header(&self) -> Result<BlockHeader, Error> {
         let commit_hash = self.raw.locate_branch(FINALIZED_BRANCH_NAME.into()).await?;
         let semantic_commit = self.raw.read_semantic_commit(commit_hash).await?;
-        let block_header: BlockHeader = serde_json::from_str(&semantic_commit.body)?;
-        Ok(block_header)
+        let commit = format::from_semantic_commit(semantic_commit).map_err(|e| anyhow!(e))?;
+        if let Commit::Block(block_header) = commit {
+            Ok(block_header)
+        } else {
+            Err(anyhow!(
+                "repository integrity broken; `finalized` branch is not on a block"
+            ))
+        }
     }
 
     /// Returns the reserved state from the `finalized` branch.
     pub async fn get_reserved_state(&self) -> Result<ReservedState, Error> {
         self.raw.read_reserved_state().await.map_err(|e| anyhow!(e))
     }
 
+    /// Cleans all the outdated commits, remote repositories and branches.
+    ///
+    /// It will leave only
+    /// - the `finalized` branch
+    /// - the `work` branch
+    /// - the `fp` branch.
+    ///
+    /// and
+    /// - the `p` branch
+    /// - the `a-#` branch
+    /// - the `b-#` branch
```

Comment:
```suggestion
    /// - the `p` branches
    /// - the `a-#` branches
    /// - the `b-#` branches
```

Reply:
note that `p` branch is unique. updated.

Reply:
oh right, understood